rpcclient/infrastructure.go

```diff
@@ -1524,3 +1557,65 @@ func (c *Client) BackendVersion() (BackendVersion, error) {
 
 	return *c.backendVersion, nil
 }
+
+// Batch makes batch requests
+func (c *Client) Batch() *Client {
+	c.batch = true //copy the client with changed batch setting
+	c.start()
+	return c
+}
+
+func (c *Client) sendAsync() FutureGetBulkResult {
+	// convert the array of marshalled json requests to a single request we can send
+	responseChan := make(chan *response, 1)
+	marshalledRequest := []byte("[")
+	for iter := c.batchList.Front(); iter != nil; iter = iter.Next() {
+		request := iter.Value.(*jsonRequest)
+		marshalledRequest = append(marshalledRequest, request.marshalledJSON...)
+		marshalledRequest = append(marshalledRequest, []byte(",")...)
+	}
+	if len(marshalledRequest) > 0 {
+		marshalledRequest = marshalledRequest[:len(marshalledRequest)-1]
+	}
+	marshalledRequest = append(marshalledRequest, []byte("]")...)
+	request := jsonRequest{
+		id:             c.NextID(),
+		method:         "",
+		cmd:            nil,
+		marshalledJSON: marshalledRequest,
+		responseChan:   responseChan,
+	}
+	c.sendPost(&request)
+	return responseChan
+}
+
+// Send sends batch requests
```

Comment:
Needs some explanation on what this does, especially stating that this function sends out the actual RPC request over the wire.

Reply:
Attempted to clarify in: 9c315730